tests/integration/test_charm.py

```diff
@@ -332,6 +337,215 @@ async def test_remove_postgresql_metastore_relation(ops_test: OpsTest, test_pod,
     assert num_tables == 0
 
 
+@pytest.mark.abort_on_fail
+async def test_enable_authentication(ops_test: OpsTest, charm_versions):
+    """Test the the behavior of charm when authentication is enabled."""
+    logger.info("Waiting for postgresql-k8s and kyuubi-k8s apps to be idle and active...")
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME, charm_versions.postgres.application_name], timeout=1000, status="active"
+    )
+
+    logger.info("Integrating kyuubi-k8s charm with postgresql-k8s charm over auth-db endpoint...")
+    await ops_test.model.integrate(charm_versions.postgres.application_name, f"{APP_NAME}:auth-db")
+
+    logger.info("Waiting for postgresql-k8s and kyuubi-k8s charms to be idle...")
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME, charm_versions.s3.application_name], timeout=1000
```

Comment:
**issue** you are looking for s3 app in idle instead of `postgresql`

Reply:
Good catch, thanks @welpaolo !